aten/src/ATen/UndefinedType.cpp

```diff
@@ -8,6 +8,9 @@ UndefinedType::UndefinedType()
 ScalarType UndefinedType::scalarType() const {
   return ScalarType::Undefined;
 }
+caffe2::TypeMeta UndefinedType::typeMeta() const {
+  AT_ERROR("typeMeta not defined for UndefinedType");
```

Comment:
this isn't correct, there's a properly defined TypeMeta corresponding to ScalarType::Undefined (see scalarTypeToTypeMeta).